lib/public/Files/DavUtil.php

```diff
@@ -75,8 +78,19 @@ public static function getDavPermissions(FileInfo $info): string {
 		if ($info->isUpdateable()) {
 			$p .= 'NV'; // Renameable, Moveable
 		}
+
+		// since we always add update permissions for the root of movable mounts (and thus shares)
+		// we need to check the shared cache item directly to determine if it's writable
+		$storage = $info->getStorage();
+		if ($info->getInternalPath() === '' && $storage->instanceOfStorage(SharedStorage::class)) {
```

Comment:
```suggestion
		/** @psalm-suppress UndefinedClass files_sharing may be disabled but this will still work as expected */
		if ($info->getInternalPath() === '' && $storage->instanceOfStorage(SharedStorage::class)) {
```

Add a suppress like this then, I do not see a better way. And `::class` does not actually complains on non-existing classes so it should be fine. Maybe still test this with files_sharing disabled to be sure.

A clean fix would be to add a new interface for this, like IWritableAwareStorage, and change the code for:
```suggestion
		if ($storage->instanceOfStorage(IWritableAwareStorage::class)) {
			$isWritable = $storage->isWritable($info);
```